flake.nix

```diff
@@ -2,26 +2,30 @@
 {
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs";
-    flake-utils.url = "github:numtide/flake-utils";
+
+    # <https://github.com/nix-systems/nix-systems>
+    systems.url = "github:nix-systems/default";
   };
 
-  outputs = {
-    self,
-    nixpkgs,
-    flake-utils,
-  }:
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = nixpkgs.legacyPackages.${system};
-      libselinuxPath = with pkgs;
-        lib.makeLibraryPath [
-          libselinux
-        ];
-      libaclPath = with pkgs;
-        lib.makeLibraryPath [
-          acl
-        ];
+  outputs = inputs: let
+    eachSystem = inputs.nixpkgs.lib.genAttrs (import inputs.systems);
+    pkgsFor = inputs.nixpkgs.legacyPackages;
+  in {
+    devShells = eachSystem (
+      system: let
+        inherit (pkgsFor.${system}) lib;
+
+        libselinuxPath = with pkgsFor.${system};
+          lib.makeLibraryPath [
+            libselinux
+          ];
 
-      build_deps = with pkgs; [
+        libaclPath = with pkgsFor.${system};
+          lib.makeLibraryPath [
+            acl
+          ];
+
+        build_deps = with pkgsFor.${system}; [
           clang
           llvmPackages.bintools
           rustup
```

Comment:
Ideally rustc and cargo would be provided by Nix directly, but that might be too big of a change for this PR.

Reply:
It's true it may be redundant to have both rustup and nix managing the rustc compiler, but I agree that should be done on a following PR